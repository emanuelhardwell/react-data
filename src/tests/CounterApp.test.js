//  1.- crear las siguientes pruebas para el <CounterApp/>
//  debe de mostrar <CounterApp/> correctamente (hacer match con un snapshot)
//  y sus valores por defecto

import { shallow } from "enzyme";
import CounterApp from "../CounterApp";

// Debe de mostrar el valor por defecto de 100
// usar el wrapper.find, tomando el elemento html donde se muestra el valor del contador
describe("Pruebas en el archivo CounterApp.js", () => {
  test("should hacer match el sanpshot", () => {
    const wrapper = shallow(<CounterApp />);

    expect(wrapper).toMatchSnapshot();
  });

  test(" Debe de mostrar el valor por defecto de 100", () => {
    const wrapper = shallow(<CounterApp value={100} />);

    const res = wrapper.find("h3").text();
    expect(res).toBe("100");
  });
});
