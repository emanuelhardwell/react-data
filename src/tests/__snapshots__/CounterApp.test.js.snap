// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el archivo CounterApp.js should hacer match el sanpshot 1`] = `
<div>
  <h1>
     CounterApp 
  </h1>
  <h2>
    counter: 
  </h2>
  <h3>
    10
  </h3>
  <button
    onClick={[Function]}
  >
     +1 
  </button>
  <button
    onClick={[Function]}
  >
     -1 
  </button>
  <button
    onClick={[Function]}
  >
     reset 
  </button>
</div>
`;
